streams/streams-scala/src/test/scala/org/apache/kafka/streams/scala/SerdesUnitTest.scala

```diff
@@ -0,0 +1,35 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.scala
+
+import org.apache.kafka.streams.processor.internals.testutil.LogCaptureAppender
+import org.apache.kafka.streams.kstream.WindowedSerdes.TimeWindowedSerde
+import org.junit.Assert.assertFalse
+import org.junit.Test
+
+class SerdesUnitTest {
+
+  @Test
+  def shouldLogMessageWhenTimeWindowedSerdeIsUsed(): Unit = {
+
+    Serdes.timeWindowedSerde(new TimeWindowedSerde[String]())
+    val appender = LogCaptureAppender.createAndRegister()
+    val warning = appender.getMessages()
+    assertFalse("There should be a warning about TimeWindowedDeserializer", warning.isEmpty)
```

Comment:
Not sure if this is the most secure check - went through a few different iterations but got type issues with all of them except this one. If someone has a better idea, lmk

Reply:
This test also appears to fail locally since it uses the constructor we deprecate in this PR and from a quick search it doesn't look like scala suppresses these warnings as easily as Java. Any thoughts on if this test is vital / a way to make this work? cc @guozhangwang

Reply:
Personally I'm not a big fan to add tests coverage for deprecated functions to make sure the expected warning is generated :) Also, is this test correct? Since currently the `Serdes.timeWindowedSerde` itself would not log any warning, and we would only log when the `deserialize` is called right (see my other comment)? 

Anyways, I'm actually okay if you would like to just remove this test. But on the other hand we should try to avoid a WARN at a per-record granularity too.

Reply:
Cool, I'm going to go ahead and take the test out. Thanks!